hi-ml-cpath/testhisto/testhisto/preprocessing/test_tiff_conversion.py

```diff
@@ -0,0 +1,215 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import math
+import numpy as np
+import pytest
+
+from pathlib import Path
+from monai.data.wsi_reader import WSIReader
+from health_cpath.datasets.panda_dataset import PandaDataset
+from health_cpath.preprocessing.loading import WSIBackend
+from health_cpath.preprocessing.tiff_conversion import AMPERSAND, UNDERSCORE, ConvertWSIToTiffd, WSIFormat
+from health_cpath.utils.naming import SlideKey
+from health_cpath.utils.tiff_conversion_config import TiffConversionConfig
+from typing import Any, List, Dict, Optional
+from unittest.mock import MagicMock
+
+
+WSISamplesType = List[Dict[SlideKey, Any]]
+
+
+@pytest.fixture
+def wsi_samples(mock_panda_slides_root_dir: Path) -> WSISamplesType:
+    wsi_root_dir = mock_panda_slides_root_dir / "train_images"
+    wsi_filenames = ["_1.tiff", "_2.tiff"]
+    return [
+        {
+            SlideKey.SLIDE_ID: wsi_filename.split(".")[0],
+            SlideKey.IMAGE: wsi_root_dir / wsi_filename,
+        }
+        for wsi_filename in wsi_filenames
+    ]
+
+
+@pytest.mark.parametrize("replace_ampersand_by", [UNDERSCORE, ""])
+@pytest.mark.parametrize("src_format", [WSIFormat.NDPI, WSIFormat.TIFF])
+def test_get_tiff_path(src_format: WSIFormat, replace_ampersand_by: str) -> None:
+    output_folder = Path("foo")
+    transform = ConvertWSIToTiffd(
+        output_folder=output_folder, image_key=SlideKey.IMAGE, src_format=src_format,
+        replace_ampersand_by=replace_ampersand_by,
+    )
+    src_path = Path(f"root/h&e_foo.{src_format}")
+    formated_path = transform.get_tiff_path(src_path)
+    assert formated_path == output_folder / f"h{replace_ampersand_by}e_foo.tiff"
+
+
+def test_base_objective_power(wsi_samples: WSISamplesType) -> None:
+    target_mag = 2.5
+    transform = ConvertWSIToTiffd(
+        output_folder=Path("foo"),
+        image_key=SlideKey.IMAGE,
+        src_format=WSIFormat.TIFF,
+        target_magnifications=[target_mag],
+        base_objective_power=None,
```

Comment:
Here and in other tests: you are providing arguments to ConvertWSIToTiffd that are already the default values. The tests
would be easier to read if you only provide those that matter to your test. even output_folder could have a default
value that is "." to further simplify.

Reply:
I am a bit too cautious with default values, if the default values change for some reason the test might break! It should be fairly easy to fix but it took me one hour once to figure out why my test didn't pass (it was because of the default values changing). I usually use default values that are useful to run the script with minimum command line arguments (hence the tile size deault to 512 that worked best for ndpi conversion) but try to reset these in the tests, but agreed I can simplify tests that only check exceptions for example.